### PR TITLE
CLDSRV-174 bump arsenal to error breaking change

### PR DESCRIPTION
**Pull request by jbertran**
### Motivation and context

Typescript migration introduced a breaking change in arsenal errors. This PR updates Arsenal and all Arsenal error uses to conform with the new API.

## Checklist

- [x] bump Arsenal
- [x] unit tests
- [x] functional tests

### CODE REVIEW

**Comment by jbertran** [comment]
Unit tests are all green :partying_face:

Functional tests not so much, but no details except we're timeouting on `s3server` startup.

Running `yarn start`, we seem to be crashing on establishing the connection to LevelDB (`MetadataFileClient`), which makes all workers immediately give up.

```
/home/jbertran/git/work/object/cloudserver/lib/metadata/bucketfile/backend.js:38
            this.metastore = this.mdDB.openSub(METASTORE);
                                       ^

TypeError: Cannot read properties of undefined (reading 'openSub')
    at /home/jbertran/git/work/object/cloudserver/lib/metadata/bucketfile/backend.js:38:40
    at /home/jbertran/git/work/object/cloudserver/node_modules/arsenal/build/lib/storage/metadata/file/MetadataFileClient.js:55:39
    at /home/jbertran/git/work/object/cloudserver/node_modules/arsenal/build/lib/network/rpc/rpc.js:141:24
    at injectedCallback (/home/jbertran/git/work/object/cloudserver/node_modules/arsenal/node_modules/async/dist/async.js:4717:30)
    at Socket.wrapCb (/home/jbertran/git/work/object/cloudserver/node_modules/arsenal/build/lib/network/rpc/rpc.js:72:13)
    at Socket.onack (/home/jbertran/git/work/object/cloudserver/node_modules/socket.io-client/lib/socket.js:319:9)
    at Socket.onpacket (/home/jbertran/git/work/object/cloudserver/node_modules/socket.io-client/lib/socket.js:244:12)
    at Manager.<anonymous> (/home/jbertran/git/work/object/cloudserver/node_modules/component-bind/index.js:21:15)
    at Manager.Emitter.emit (/home/jbertran/git/work/object/cloudserver/node_modules/component-emitter/index.js:145:20)
    at Manager.ondecoded (/home/jbertran/git/work/object/cloudserver/node_modules/socket.io-client/lib/manager.js:349:8)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Next step is diving into the call stack to understand why we're unable to get this started.

File: lib/api/apiUtils/object/versioning.js [hunk 1]
```diff
@@ -292,7 +292,7 @@ function versioningPreprocessing(bucketName, bucketMD, objectKey, objMD,
                         // it's possible there was a concurrent request to
                         // delete the null version, so proceed with putting a
                         // new version
-                        if (err === errors.NoSuchKey) {
+                        if (err === errors.is.NoSuchKey) {
```
Comment (XinLi0207) [inline lib/api/apiUtils/object/versioning.js:295]: should be `if(err.is.NoSuchKey)` ?

File: lib/api/multiObjectDelete.js [hunk 2]
```diff
@@ -386,11 +386,11 @@ function multiObjectDelete(authInfo, request, log, callback) {
             return vault.checkPolicies(requestContextParams, authInfo.getArn(),
                 log, (err, authorizationResults) => {
                     // there were no policies so received a blanket AccessDenied
-                    if (err && err.AccessDenied) {
+                    if (err && err.is.AccessDenied) {
                         objects.forEach(entry => {
                             errorResults.push({
                                 entry,
-                                error: errors.AccessDenied });
+                                error: errors.is.AccessDenied });
```
Comment (XinLi0207) [inline lib/api/multiObjectDelete.js:393]: maybe no need `is` here?

**Comment by jbertran** [comment]
We'll do this in 2 steps:
1. `feature/CLDSRV-174-use-armory-work` "just" bumps Arsenal _and_ merges @miniscruff 's CLDSRV-162 work
2. CLDSRV-181 will update to the new error usage

**Pull request closed without merging**